src/components/Checkbox.js

```diff
@@ -1,84 +1,74 @@
 import cx from 'classnames';
 import PropTypes from 'prop-types';
-import React from 'react';
+import React, { useRef, useEffect, useState } from 'react';
 
 import { generateUUID } from '../helpers/utility';
 import { config } from '../helpers/config';
 
-class Checkbox extends React.Component {
-  checkboxLabelRef = React.createRef();
-
-  id = `${this.props.name}-${generateUUID()}`; // eslint-disable-line react/destructuring-assignment
-
-  state = {
-    checked: this.props.isChecked, // eslint-disable-line react/destructuring-assignment
-  };
-
-  componentWillReceiveProps(newProps) {
-    const { isChecked } = this.props;
-
-    if (newProps.isChecked !== isChecked) {
-      this.setState({
-        checked: newProps.isChecked,
-      });
-    }
-  }
-
-  toggleChecked = () => {
-    const { onChange, name, value } = this.props;
-    const { checked } = this.state;
+const Checkbox = ({
+  className,
+  disabled,
+  isChecked,
+  label,
+  name,
+  onChange,
+  value,
+}) => {
+  const checkboxLabelRef = useRef();
+  const [checked, setChecked] = useState(isChecked);
+
+  const id = `${name}-${generateUUID()}`;
+
+  useEffect(
+    () => {
+      setChecked(isChecked);
+    },
+    [isChecked],
+  );
+
+  const toggleChecked = () => {
+    setChecked(!checked);
```

Comment:
Is this internal state needed? Could you use the isChecked prop instead of assigning it to an internal state variable? This seems to be both a controlled and uncontrolled component.